src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -416,9 +417,36 @@ public InnReachResponseDTO finalCheckIn(String trackingId, String centralCode, B
 
     transaction.setState(FINAL_CHECKIN);
 
+    removeFinalCheckInFieldsFromItem(transaction);
+    removeFinalCheckInFieldsFromHold(transaction);
+
     return success();
   }
 
+  private void removeFinalCheckInFieldsFromItem(InnReachTransaction transaction) {
+    var itemBarcode = transaction.getHold().getFolioItemBarcode();
+    var itemDTO = itemService.findItemByBarcode(itemBarcode).orElseThrow(
+      () -> new EntityNotFoundException("FOLIO item record is not found for barcode: " + itemBarcode));
+
+    itemDTO.setBarcode(null);
+
+    itemService.update(itemDTO);
+  }
```

Comment:
there is already logic implemented for clearing Patron Hold inventory records info. I assume here we should apply the same approach. Clarify with PO if we need to remove a callNumber of item/holding

https://github.com/folio-org/mod-inn-reach/blob/b33d7e9f9966975ec8ad51f59abf57f665e85db8/src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java#L225-L226